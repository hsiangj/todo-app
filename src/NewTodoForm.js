import React, {useState} from "react";
import {v4 as uuid} from 'uuid';

const NewTodoForm = ({addTodo}) => {
  const [task, setTask] = useState('');
  const handleChange = (e) => {
    setTask(e.target.value);
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({task, id: uuid()});
    setTask('');
  }

  return(
    <form onSubmit={handleSubmit}>
      <label htmlFor="todo">New task: </label>
      <input
        id="todo" 
        type="text"
        name="task"
        value={task}
        onChange={handleChange}
      />
      <button>Add</button>
    </form>
  )
}

export default NewTodoForm;