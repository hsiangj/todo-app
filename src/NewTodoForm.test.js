import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it("renders without crashing", function(){
  render(<NewTodoForm />);
});

it("matches snapshot", function(){
  const {asFragment} = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it("runs addTodo function on form submit", function(){
  const addMock = jest.fn();
  const {getByText} = render(<NewTodoForm addTodo={addMock} />);
  const addButton = getByText('Add');
  fireEvent.click(addButton);
  expect(addMock).toHaveBeenCalled();
});