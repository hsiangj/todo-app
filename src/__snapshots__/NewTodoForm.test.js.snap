// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      New task: 
    </label>
    <input
      id="todo"
      name="task"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
